### PR TITLE
cluster-autoscaler Azure e2e: move Helm to ginkgo

### PR DESCRIPTION
**Pull request by nojnhuh**
#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:

This PR moves the `helm install` command for cluster-autoscaler Azure e2e tests into the testing framework to better manage cluster-autoscaler config between test cases.

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: cluster-autoscaler/cloudprovider/azure/test/e2e/e2e_suite_test.go [hunk 1]
```diff
@@ -53,11 +77,41 @@ var _ = BeforeSuite(func() {
 	vmss, err = armcompute.NewVirtualMachineScaleSetsClient(os.Getenv("AZURE_SUBSCRIPTION_ID"), azCred, nil)
 	Expect(err).NotTo(HaveOccurred())
 
-	k8sConfig := genericclioptions.NewConfigFlags(false)
-	restConfig, err := k8sConfig.ToRESTConfig()
+	restConfig, err := helmEnv.RESTClientGetter().ToRESTConfig()
```
Comment (jackfrancis) [inline cluster-autoscaler/cloudprovider/azure/test/e2e/e2e_suite_test.go:80]: Eventually we're going to want to be able to switch between installing cluster-autoscaler on the workload cluster (AKS and Azure self-managed k8s scenarios) or the CAPI mgmt cluster (Cluster API provider scenario). I think the way this is now should allow for that evolution fairly easily, but just FYI.

**Comment by nojnhuh** [comment]
Test is green!
/hold cancel

**Pull request merged**